quinn-proto/src/connection/mod.rs

```diff
@@ -483,7 +483,7 @@ impl Connection {
                     SpaceId::Data,
                     "PATH_CHALLENGE queued without 1-RTT keys"
                 );
-                buf.reserve(self.path.current_mtu() as usize);
+                buf.reserve(prev_path.current_mtu() as usize);
```

Comment:
We're actually always sending a fixed-size packet here; see the `pad_to` call below, and the absence of any frames other than the `PATH_CHALLENGE`. Maybe `reserve(MIN_INITIAL_SIZE)` would be clearer still?